app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddSmoothness.kt

```diff
@@ -0,0 +1,71 @@
+package de.westnordost.streetcomplete.quests.smoothness
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.ALL_ROADS
+import de.westnordost.streetcomplete.data.meta.ANYTHING_PAVED
+import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChangesBuilder
+import de.westnordost.streetcomplete.quests.surface.Surface
+
+class AddSmoothness : OsmFilterQuestType<SmoothnessAnswer>() {
+
+    // maybe exclude service roads? or driveways?
+    override val elementFilter = """
+        ways with highway
+        and highway ~ ${(ALL_PATHS_EXCEPT_STEPS + ALL_ROADS).joinToString("|")}
+        and surface ~ ${SURFACES_FOR_SMOOTHNESS.joinToString("|")}
+        and access !~ private|no
+        and (!conveying or conveying = no)
+        and (!indoor or indoor = no)
+        and !cycleway:surface and !footway:surface
+        and (
+          !smoothness
+          or smoothness older today -4 years
+        )
+    """
+    /* resurvey:
+     * what it more/less likely to change?
+     * paved and good/intermediate might stay that way for a long time
+     * excellent might soon change to good
+     * bad roads might get repaired
+     * unpaved roads are likely to change anyway
+     * metal should stay the way it is for a long time
+     *  */
```

Comment:
Alright, so keep it at 4 years?

Could this interfere with surface resurvey?
As far as I understand surface is asked every 8 years for paved surfaces. If smoothness is asked more often than surface, SC will never add a check date for surface and will not re-ask for surface.

Reply:
Yes, absolutely it does interfere. Only if a check date for the surface tag is set on the way, it won't. StreetComplete does not set a check date on something if the value changed in order to be frugal with `check_date` tagging because some people on the tagging mailing list requested this (including @matkoniecz ).
So, nothing we can do about that, really.

Reply:
Ok, so currently paved surfaces will not be resurveyed, and for other surfaces it depends on quest priority whether surface or smoothness will be resurveyed.